opencog/moses/main/populate_atomspace.cc

```diff
@@ -7,28 +7,28 @@
 
 namespace opencog { namespace moses {
 
-void populate(AtomSpace *as, ITable &itable) {
+void populate(AtomSpace *as, const ITable &itable) {
 
 	int n_columns = itable.get_types().size();
 	for (int i = 0; i < n_columns; i++) {
 		id::type_node col_type = itable.get_types().at(i);
-		vertex_seq col = itable.get_column_data(i);
-		Handle feature = createNode(SCHEMA_NODE,itable.get_labels().at(i));
-		ValuePtr vtr = vertex_seq_to_value(col, col_type);
+		const vertex_seq &col = itable.get_column_data(i);
+		const Handle &feature = createNode(SCHEMA_NODE,itable.get_labels().at(i));
```

Comment:
I know that passing std::vector by ref makes a difference in performance. I have less/no such experience for shared pointer (I mean they are fairly like object), but it doesn't hurt to be there, so I'm letting it.